commit e89033fae0546c36010046c5e27d6a7371638308 (refs/changes/15/3215/4)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-16T20:39:51+01:00 (1 year, 9 months ago)
    
    TOOLS-1973 sdcadm up moray throws an exception when a VM never made it from SAPI into VMAPI
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/procedures/update-moray-v2.js b/lib/procedures/update-moray-v2.js
index ded1934..6a83962 100644
--- a/lib/procedures/update-moray-v2.js
+++ b/lib/procedures/update-moray-v2.js
@@ -12,6 +12,7 @@ var assert = require('assert-plus');
 var sprintf = require('extsprintf').sprintf;
 var util = require('util');
 var vasync = require('vasync');
+var verror = require('verror');
 
 var errors = require('../errors');
 var common = require('../common');
@@ -130,6 +131,14 @@ UpdateMorayV2.prototype.execute = function morayv2Execute(opts, cb) {
                             uuid: tmpInst.uuid
                         }, function (vmErr, vm) {
                             if (vmErr) {
+                                if (verror.hasCauseWithName(
+                                    vmErr, 'ResourceNotFoundError')) {
+                                    ctx.tmpInstToRemove =
+                                        ctx.tmpInstToRemove || [];
+                                    ctx.tmpInstToRemove.push(tmpInst.uuid);
+                                    nextVm();
+                                    return;
+                                }
                                 nextVm(new errors.SDCClientError(vmErr,
                                     'vmapi'));
                                 return;
@@ -173,7 +182,8 @@ UpdateMorayV2.prototype.execute = function morayv2Execute(opts, cb) {
                     inputs: ctx.tmpInstToRemove,
                     func: function removeFromSapi(inst, nextInst) {
                         sdcadm.sapi.deleteInstance(inst, function (remErr) {
-                            if (remErr) {
+                            if (remErr && !verror.hasCauseWithName(remErr,
+                                'ResourceNotFoundError')) {
                                 nextInst(new errors.SDCClientError(remErr,
                                     'sapi'));
                                 return;
